pkg/controller/backup/backup_controller.go

```diff
@@ -263,3 +256,348 @@ func (c *Controller) enqueueBackup(obj interface{}) {
 	}
 	c.queue.Add(key)
 }
+
+// detectBackupJobOrPodFailure detect backup job or pod failure.
+// it will record failure info to backup status, it is to realize reentrant logic for spec.backoffRetryPolicy. so it only record snapshot backup failed now.
+func (c *Controller) detectBackupJobOrPodFailure(backup *v1alpha1.Backup) (bool, string, error) {
```

Comment:
1. detectBackupJobOrPodFailure =>  detectBackupFailure
2. detectBackupJobOrPodFailure not just detect failure, but also do recording, please comply with single responsibility principle when you writing code

Reply:
1. i'm afraid `detectBackupFailure` may make others think it detect backup cr's failure which may be normal backup failure. so i prefer to add the describe of job or pod to the function name.
2. record infomation in backup cr is  the main way to achieve reentrant, include retry after detect failure here, so maybe it is not necessary to specify record it in the function name.